codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/integration/HttpRpcProtocolGenerator.java

```diff
@@ -249,7 +249,11 @@ private void generateOperationDeserializer(GenerationContext context, OperationS
             writer.write("");
 
             writer.openBlock("if response.StatusCode < 200 || response.StatusCode >= 300 {", "}", () -> {
-                writer.write("return out, metadata, $L(response)", errorFunctionName);
+                writer.write("deserMetadata, err := $L(response)", errorFunctionName);
```

Comment:
These deserializer functions probably should take `metadata` as a pointer input, `&metadata`, instead of returning a new one that will need to be merged into the stack's metadata.

Reply:
I didn't do that as that change would in the smithy-go client and would force all smithy-go clients to take metadata as input for top level error deser func

I think returning a empty interface{} allows us to keep it more generic, especially because we have a `processErrorResponseMetadata` method follow by error deserialization.

With the functions returning an empty interface, we can keep aws specific implementation separate and not dictate every other Smithy-go client.